src/account.ts

```diff
@@ -95,10 +93,14 @@ export class Account {
 
     /**
      * Returns the state of a NEAR account
-     * @returns {Promise<AccountState>}
+     * @returns {Promise<AccountView>}
      */
-    async state(): Promise<AccountState> {
-        return await this.connection.provider.query(`account/${this.accountId}`, '');
+    async state(): Promise<AccountView> {
+        return this.connection.provider.query<AccountView>({
+            request_type: 'view_account',
+            account_id: this.accountId,
+            finality: 'optimistic'
```

Comment:
are we sure that default was optimistic?

Reply:
I believe so:

* using the older style falls back to this: https://github.com/near/nearcore/blob/master/chain/jsonrpc-primitives/src/types/query.rs#L146
* `Finality::None` is this https://github.com/near/nearcore/blob/master/core/primitives/src/types.rs#L23